### ISSUE TITLE
Dockerhub not updated for 3.10.2

### ISSUE THREAD

**Issue by cwhits** [pducharme/UniFi-Video-Controller#133]
on docker hub, `latest` is still pointing to 3.10.1 and there's no tag for 3.10.2. Can someone give the docker build a kick? @pducharme?

**Comment by xboxhacks** [comment]
Yep also have the issue, latest = 3.10.1

**Comment by jpetidis** [comment]
Same issue here :(

**Comment by phedders** [comment]
Also current 3.10.2 Dockerfile does not build because Ubiquiti changed the deb path.... and also changed the main unifi-video script...
fixed in b94f4195a9f936c2c2d57f283fbddc7a3fe03781 - see pull request.

**Comment by BlamKiwi** [comment]
Same issue here. Forking the repo directly for now to fix some deployments.

**Comment by d2dyno** [comment]
I used @phedders latest fix, and pushed 3.10.2 to my dockerhub for this:
https://hub.docker.com/r/d2dyno/unifi-video (Rebased from my old builds: https://github.com/d2dyno/UniFi-Video-Controller/pull/1)

**Comment by fryfrog** [comment]
I can't upgrade from 3.10.1 -> 3.10.2 w/o losing my user database table to "corruption" (but works fine when downgraded back to 3.10.1). Is it working fine for all of you?

**Comment by cwhits** [comment]
I haven't tried. I usually remove my config dir and restore from one of the automatic backups.

**Comment by fryfrog** [comment]
I don't understand how they think that is a reasonable upgrade path :/

**Comment by fss** [comment]
@fryfrog aside of reasonable / not reasonable upgrade path, can you update the dockerhub?

**Comment by fryfrog** [comment]
The `:beta` and `:testing` tags have 3.10.2, I'd like to stick at 3.10.1 until I understand the problem and can offer a better upgrade path. :/

**Comment by fss** [comment]
OK, thanks, fair enough, also I've read lots of frustration from 3.10.2 users on UBNT forums:

https://community.ubnt.com/t5/UniFi-Video/DO-NOT-UPDATE-TO-UNIFI-VIDEO-FIRMWARE-3-10-1-or-3-10-2-OR-RISK/m-p/2755356

https://community.ubnt.com/t5/UniFi-Video/3-10-1-gt-3-10-2-Error-Upgrading-Server-Download-failed-Illegal/td-p/2746564

so it's indeed better to have a working 3.10.1, it seems ;)

**Comment by BlamKiwi** [comment]
For the purposes of maintaining the dockerhub repo it would probably be a better idea to have a "legacy" tag introduced which stays at 3.10.1.

**Comment by fryfrog** [comment]
Hey, that is actually a good point. It has a 3.10.1 tag so anyone that wants to stick to that can. I'll bump latest to 3.10.2.

**Comment by Elehiggle** [comment]
For anyone trying to update from 3.9 to 3.10.2 without much effort, do the following steps:
- Save backup
- Update Docker to Docker 3.9.12 if you haven't already
- Use the web-interface upgrade process to 3.10.2, wait until it is stuck and nothing happens for 5 minutes
- Shut down docker, delete the docker
- Upgrade docker to 3.10.2
- Enjoy

Worked for me, camera upgraded their firmware on the last step and are connectable. I didn't have to delete any data folder.

tl;dr: Be on 3.9.12, backup, use web-upgrade to 3.10.2, wait until stuck, delete docker and upgrade to 3.10.2

**Comment by fryfrog** [comment]
https://community.ubnt.com/t5/UniFi-Video/3-10-1-gt-3-10-2-can-not-log-in/m-p/2765592/highlight/true#M122148

I finally figured out the problem. And why a backup / restore has been fixing it for people.
